File: posesor-webapi/src/test/java/net/posesor/expenses/Given.java
package net.posesor.expenses;

import lombok.val;

import java.math.BigDecimal;
import java.util.UUID;

public class Given {

    private static int counter = 1;

    /**
     * Disable ability to create utility class.
     */
    private Given() {
    }

    public static ExpenseDocumentDto prepareFullExpenseDtoModel() {
        val entity = new ExpenseDocumentDto();
        val random = UUID.randomUUID().toString();
        entity.setSubjectName("Subject Name " + random);
        entity.setCustomerName("Customer name " + random);
        entity.setPaymentTitle("Payment title" + random);
        entity.setPaymentDate("2001-02-03");
        entity.setAmount(BigDecimal.valueOf(99));
        entity.setDescription("Description " + random);
        return entity;
    }
}

File: posesor-webapi/src/test/java/net/posesor/allocations/Given.kt
package net.posesor.allocations

import java.math.BigDecimal
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.util.*

private fun String.Companion.random(): String {
    return UUID.randomUUID().toString()
}

public fun fullValidChargeDto() : net.posesor.charges.ChargeDocumentDto {
    return net.posesor.charges.ChargeDocumentDto().apply {
        customerName = "Customer Name ${UUID.randomUUID()}"
        subjectName = "Subject Name ${UUID.randomUUID()}"
        paymentTitle = "Payment Title ${UUID.randomUUID()}"
        amount = BigDecimal.valueOf(10)
        paymentDate = LocalDate.now().format(DateTimeFormatter.ISO_LOCAL_DATE)
    }
}

